ddl/src/main/java/com/adaptivescale/rosetta/ddl/targets/kinetica/KineticaDDLGenerator.java

```diff
@@ -225,6 +223,43 @@ private Optional<String> createPrimaryKeysForTable(Table table, List<String> for
         return Optional.of("PRIMARY KEY (" + String.join(", ", primaryKeys) + ")");
     }
 
+    private List<String> getIndicesForTable(Table table) {
+        List<Index> result = table.getIndices();
+        List<String> indices = new ArrayList<>();
+        List<String> generateIndexStatement = generateIndexStatements(result.stream().findFirst().get().getColumnNames());
+        return generateIndexStatement;
+    }
+
+    public static List<String> generateIndexStatements(List<String> values) {
+        List<String> indexStatements = new ArrayList<>();
+
+        for (String value : values) {
+            String statement = generateIndexStatement(value);
+            indexStatements.add(statement);
+        }
+
+        return indexStatements;
+    }
+
+    private static String generateIndexStatement(String value) {
+        if (value.contains("@")) {
+            String[] parts = value.split("@");
+            String type = parts[0].toUpperCase();
+            String[] ids = parts[1].split(":");
+            String joinedIds = String.join(", ", ids);
+            switch (type.toLowerCase()) {
+                case "geospatial":
+                    return String.format("GEOSPATIAL INDEX (%s)", joinedIds);
```

Comment:
```
GEOSPATIAL INDEX
CHUNK SKIP
INDEX
```
maybe can be CONSTANTS for clarity

Reply:
Yes i agree this has been resolved aswell